### ISSUE TITLE
FP unnecessary-semicolon

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#14100]
### Summary

.

### Lint Name

unnecessary-semicolon

### Reproducer

I tried this code:

```rust
//@ run-pass

#![allow(irrefutable_let_patterns)]

fn _f1() -> bool {
	if let _ = return true && false {};
}

pub fn main() {}
```

I saw this happen:

```
warning: unnecessary semicolon
 --> ./tests/ui/rfcs/rfc-2497-if-let-chains/protect-precedences.rs:6:36
  |
6 |     if let _ = return true && false {};
  |                                       ^ help: remove
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_semicolon
  = note: requested on the command line with `--force-warn clippy::unnecessary-semicolon`

warning: 3 warnings emitted
```

I expected to see this happen:
removing it breaks the build
```rust
error[E0308]: mismatched types
 --> o.rs:4:34
  |
4 |     if let _ = return true && false {}
  |                                     ^^ expected `bool`, found `()`

error: aborting due to 1 previous error; 1 warning emitted
```

### Version

```text
rustc 1.86.0-nightly (ccc9ba5c3 2025-01-29)
binary: rustc
commit-hash: ccc9ba5c30c675824e9ca62b960830ff4a1858ea
commit-date: 2025-01-29
host: x86_64-unknown-linux-gnu
release: 1.86.0-nightly
LLVM version: 19.1.7
```
